lib/src/themes/theme_reader.dart

```diff
@@ -203,11 +203,26 @@ class ThemeReader {
   }
 
   LineLayout _toLineLayout(jsonLayer) {
+    List<num> dashArray = [];
+
+    // check if we need to paint line dashed
+    final dashJson = jsonLayer['paint']['line-dasharray'];
+    if (dashJson != null) {
+      // check if at least 2 values are specified (otherwise dashing useless)
+      if ((dashJson as List<num>).length >= 2) {
+        // due to spec vals must be >= 0
+        assert(dashJson[0] >= .0);
```

Comment:
rather than have this fail, please use `logger.warn()` instead. That way invalid styles don't cause the map to be blank or the application to crash.

Reply:
Done, thank you